instructor/common.py

```diff
@@ -0,0 +1,604 @@
+from django.shortcuts import render_to_response, redirect
+from django.template import RequestContext
+from django import forms
+from django.forms.models import modelformset_factory, inlineformset_factory
+from django.forms.models import modelform_factory
+from instructor import models
+from instructor import compiler
+from django.contrib.auth.models import User
+import datetime
+
+from backend.models import Assignment, StudentAssignment
+
+
+def courses(request):
+    user = request.user
+    message = ''
+    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'], can_delete=True)
+    if request.method == "POST":
+        formset = CourseFormSet(request.POST)
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.owner = request.user
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/courses_new.html',
+                              {'formset': CourseFormSet(queryset=models.Course.objects.filter(owner=user)),
+                               'message': message},
+                              context_instance=RequestContext(request))
+
+
+def course_delete(request, pk):
```

Comment:
login required, or else anyone can delete any course.  Also, don't you want some permissions checks here, so that not every teacher can delete every other teachers courses?